src/settings.cpp

```diff
@@ -841,42 +848,48 @@ directoryAddress = GetIniSetting ( IniXMLDocument, "server", "centralservaddr",
     }
     else
     {
-        // clang-format off
-// TODO compatibility to old version < 3.4.7
-if ( GetFlagIniSet ( IniXMLDocument, "server", "defcentservaddr", bValue ) )
-{
-    directoryType = bValue ? AT_DEFAULT : AT_CUSTOM;
-}
-else
-            // clang-format on
+        //### TODO: BEGIN ###//
+        // compatibility to old version < 3.4.7
+        if ( GetFlagIniSet ( IniXMLDocument, "server", "defcentservaddr", bValue ) )
+        {
+            directoryType = bValue ? AT_DEFAULT : AT_CUSTOM;
+        }
+        else
+            //### TODO: END ###//
 
             // if "directorytype" itself is set, use it (note "AT_NONE", "AT_DEFAULT" and "AT_CUSTOM" are min/max directory type here)
-            // clang-format off
-// TODO compatibility to old version < 3.8.2
-if ( GetNumericIniSet ( IniXMLDocument, "server", "centservaddrtype", static_cast<int> ( AT_DEFAULT ), static_cast<int> ( AT_CUSTOM ), iValue ) )
-{
-    directoryType = static_cast<EDirectoryType> ( iValue );
-}
-else
-            // clang-format on
+
+            //### TODO: BEGIN ###//
+            // compatibility to old version < 3.8.2
             if ( GetNumericIniSet ( IniXMLDocument,
                                     "server",
-                                    "directorytype",
-                                    static_cast<int> ( AT_NONE ),
+                                    "centservaddrtype",
+                                    static_cast<int> ( AT_DEFAULT ),
                                     static_cast<int> ( AT_CUSTOM ),
                                     iValue ) )
         {
             directoryType = static_cast<EDirectoryType> ( iValue );
         }
+        //### TODO: END ###//
+
+        else if ( GetNumericIniSet ( IniXMLDocument,
+                                     "server",
+                                     "directorytype",
+                                     static_cast<int> ( AT_NONE ),
+                                     static_cast<int> ( AT_CUSTOM ),
+                                     iValue ) )
+        {
+            directoryType = static_cast<EDirectoryType> ( iValue );
+        }
```

Comment:
The diff looks a bit strange here, but I think it's ok.

Reply:
Yeah, I had too look 3 times too here...
Looks a bit strange because of the splitting up of if else.

If I remember well this was exactly the spot where also clang-format got confused ;=))